### PR TITLE
Promotion from AMD internal branch for 2022.Q1.3

### PR DESCRIPTION
**Pull request by WenqingLiAMD**
@kuhar

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 5a5e60c51e6fac0fa11e3b1fa8dcdc7d06ff8e6a
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev b638e90ca4e6e5a6fc4f00029d62f8e064aa18eb</li>
<li>SPVGEN: dev 34c9f9a74bc1b2b9a739933923920c5eeb9aa08f</li>
<li>XGL: dev 89a4cf115b9a2972ff00e2fe4959f4f1c8e7175b</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev afe13751c21d672a992d83f7d33856958f7d2f3b</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/179431)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 37595/66653 (56.4%)</li>
<li>Failed: 0/66653 (0.0%)</li>
<li>Not Supported: 29058/66653 (43.6%)</li><li>Warnings: 0/66653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 28817/46125 (62.5%)</li>
<li><details><summary>Failed: 1/46125 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.sparse_resources.image_sparse_memory_aliasing.cube.rgba8ui.128_128_1
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 17307/46125 (37.5%)</li><li>Warnings: 0/46125 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38653/66653 (58.0%)</li>
<li>Failed: 0/66653 (0.0%)</li>
<li>Not Supported: 28000/66653 (42.0%)</li><li>Warnings: 0/66653 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/tool/llpcPipelineBuilder.h [hunk 1]
```diff
@@ -65,7 +65,6 @@ class PipelineBuilder {
 
   virtual ~PipelineBuilder() = default;
 
-  // Compiles the pipeline and performs linking.
```
Comment (kuhar) [inline llpc/tool/llpcPipelineBuilder.h:68]: This shouldn't be removed

File: tool/vfx/vfxVkSection.h [hunk 2]
```diff
@@ -452,7 +453,7 @@ class SectionGraphicsState : public Section {
 
 private:
   SectionNggState m_nggState;
-  static const unsigned MemberCount = 28;
+  static const unsigned MemberCount = 25;
```
Comment (kuhar) [inline tool/vfx/vfxVkSection.h:456]: I can see one new member added, so I'd expect this to change to **increase** to 29 instead of **decreasing** to 25.

**Review by kuhar — CHANGES_REQUESTED** [review]
shaderdb tests are failing. will this be resolved by updating other submodules?

The clang-tidy warnings seems innocous in this case (they only report issues with the khronos header which we should not modify).

File: tool/vfx/vfxVkSection.h (hunk 2 above)
Comment (WenqingLiAMD) [reply tool/vfx/vfxVkSection.h:456]: we removed several members before in internal branch, but didn't update the count, the change to 25 is for that, this count is good for open source  too.

Comment (kuhar) [reply tool/vfx/vfxVkSection.h:456]: Is this compatible with old XGL/PAL code drops? We are still on v215.

Comment (WenqingLiAMD) [reply tool/vfx/vfxVkSection.h:456]: this change is independent of xgl/pal, there should be no compatible issue.

File: llpc/tool/llpcPipelineBuilder.h (hunk 1 above)
Comment (WenqingLiAMD) [reply llpc/tool/llpcPipelineBuilder.h:68]: will fix this, thanks

**Pull request closed without merging**